solutions/cses-1640.mdx

```diff
@@ -141,3 +123,175 @@ print("IMPOSSIBLE")
 </PySection>
 
 </LanguageSection>
+
+## Solution 2 - Two Pointers
+
+By keeping two pointers, one at each end of the list. We can find the two indices in $\mathcal{O}(N)$ time.
+<LanguageSection>
+<JavaSection>
+
+```java
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.*;
+
+public class SumOfTwoValues {
+	public static void main(String[] args) {
+		FastScanner fs = new FastScanner();
+		long N, X;
+		N = fs.nextInt();
+		X = fs.nextInt();
+
+		List<Pair> a = new ArrayList<Pair
+						  
+		// append the element and its index
+		for (int i = 0; i < N; i++) {
+			long x = fs.nextInt();
+			a.add(new Pair(x, i + 1));
+		}
+		Collections.sort(a);
+		int i = 0, j = (int) N - 1;
+		while (i < j) {
+			// adjust the left and right pointers.
+			if (a.get(i).f() + a.get(j).f() > X)
+				j--;
+			else if (a.get(i).f() + a.get(j).f() < X)
+				i++;
+			else if (a.get(i).f() + a.get(j).f() == X) {
+				System.out.println(a.get(i).s() + " " + a.get(j).s());
+				return;
+			}
+		}
+		System.out.println("IMPOSSIBLE");
+	}
+
+	static void sort(int[] a) {
+		ArrayList<Integer> l = new ArrayList<>();
+		for (int i : a)
+			l.add(i);
+		Collections.sort(l);
+		for (int i = 0; i < a.length; i++)
+			a[i] = l.get(i);
+	}
+	//BeginCodeSnip{FastScanner}
+	static class FastScanner {
+		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+		StringTokenizer st = new StringTokenizer("");
+
+		String next() {
+			while (!st.hasMoreTokens())
+			try {
+				st = new StringTokenizer(br.readLine());
+			} catch (IOException e) {
+				e.printStackTrace();
+			}
+			return st.nextToken();
+		}
+
+		int nextInt() {
+			return Integer.parseInt(next());
+		}
+
+		int[] readArray(int n) {
+			int[] a = new int[n];
+			for (int i = 0; i < n; i++)
+			a[i] = nextInt();
+			return a;
+		}
+
+		long nextLong() {
+			return Long.parseLong(next());
+		}
+	}
+	//EndCodeSnip
+
+	static class Pair implements Comparable<Pair> {
+		long x, y;
```

Comment:
Why not just name the instance variables `f` and `s` so that you dont need to make getter methods to access them?

Reply:
Also, maybe name them `val` and `index` so that they are descriptive of what they store

Reply:
> Also, maybe name them `val` and `index` so that they are descriptive of what they store

ok, im not too good with java, but this means that i can just call `.val` and `.index` right?

Reply:
yep